envoy-control-core/src/main/resources/lua/ingress_rbac_logging.lua

```diff
@@ -23,9 +23,20 @@ function envoy_on_response(handle)
     end
 
     local lua_metadata = handle:streamInfo():dynamicMetadata():get("envoy.filters.http.lua") or {}
-    local client_name = lua_metadata["request.info.client_name"] or ""
+    local client_name = nil
     local path = lua_metadata["request.info.path"] or ""
-    local protocol = handle:connection():ssl() == nil and "http" or "https"
+    local protocol = "http"
+    if handle:connection():ssl() ~= nil then
+        protocol = "https"
+        local uriSanPeerCertificate = handle:streamInfo():downstreamSslConnection():uriSanPeerCertificate()
+        if uriSanPeerCertificate[1] ~= nil then
+            client_name = string.match(uriSanPeerCertificate[1], "://([a-zA-Z0-9-_.]+)")
+        end
+    end
+    if client_name == "" or client_name == nil then
+        client_name = lua_metadata["request.info.client_name"] or ""
+    end
+
```

Comment:
This is taking a client name either from a header or certificate and uses it for logging. We'd still need a client name from the certificate in metadata in order to pass it to a service.
I think extracting a name from the certificate should be in `envoy_on_request` handler. That way we can add a legit identity header for the service to consume. 
For some responsibility segregation and to simplify tests, I'd create a separate filter that adds this legit identity header before RBAC filter than add that header name to `client_identity_headers` and reuse the existing mechanism.

Reply:
I think current `client_identity_headers` mechanism is too simple to use it at it is. For example we will not be able to express a logic described here

Reply:
Also extracting data from certificate in `envoy_on_response` is better performance-wise. We only do it to requests, that were denied by RBAC filter, not all requests.

Reply:
But does not address providing trusted client name to upstream.

Reply:
Regarding not being able to handle `not trusted`, that's UX discussion. I'd log all specified identity headers and decide how to present it reader-side.

Reply:
>  I'd log all specified identity headers and decide how to present it reader-side

I agree, we could simply pass raw data in the output let the reader handle how to interpret it. But it will require more work and we probably will have compatibility issues between between EC and log reader. And we probably would need to improve that "hand made" JSONish format the lua script produces.
But I agree - it is definitely a better idea.

> But does not address providing trusted client name to upstream.
That's right, but I think we didn't even decide how we actually will implement that. There are some ideas to implement it without even using lua scripts. But probably we will use Lua for the task to be honest. So that's true, it's probably better to do it on the request side.